test/unit/math/opencl/kernel_generator/binary_operation_test.cpp

```diff
@@ -0,0 +1,209 @@
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/kernel_generator.hpp>
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/copy.hpp>
+#include <Eigen/Dense>
+#include <gtest/gtest.h>
+#include <string>
+
+using Eigen::MatrixXd;
+using Eigen::MatrixXi;
+using stan::math::matrix_cl;
+
+#define EXPECT_MATRIX_NEAR(A, B, DELTA) \
+  for (int i = 0; i < A.size(); i++)    \
+    EXPECT_NEAR(A(i), B(i), DELTA);
+
+TEST(MathMatrixCL, addition_test) {
+  MatrixXd m1(3, 3);
+  m1 << 1, 2.5, 3, 4, 5, 6.3, 7, -8, -9.5;
+  MatrixXd m2(3, 3);
+  m2 << 10, 100, 1000, 0, -10, -12, 2, 4, 8;
+
+  matrix_cl<double> m1_cl(m1);
+  matrix_cl<double> m2_cl(m2);
+
+  auto tmp = m1_cl + m2_cl;
```

Comment:
Kind of confused here, doesn't `operator+` for `matrix_cl`s just call the code in `add.hpp`?

Reply:
Or is because that `operator+` in `addition_` is more generic so it's the one that is accepted?

Reply:
One from`add.hpp` is more specific so that one would be chosen if both were defined. But I am not including `add.hpp` into these tests. Once kernel generator is in we can remove `add.hpp` `subtract.hpp` and some parts from `multiply.hpp`.